src/Bridge/XDBCBridgeHelper.h

```diff
@@ -0,0 +1,265 @@
+
+
+template <typename BridgeHelperMixin>
+class XDBCBridgeHelper : public IXDBCBridgeHelper
+{
+
+public:
+    static constexpr inline auto DEFAULT_PORT = BridgeHelperMixin::DEFAULT_PORT;
+    static constexpr inline auto COL_INFO_HANDLER = "/columns_info";
+    static constexpr inline auto IDENTIFIER_QUOTE_HANDLER = "/identifier_quote";
+    static constexpr inline auto SCHEMA_ALLOWED_HANDLER = "/schema_allowed";
+
+    XDBCBridgeHelper(
+            const Context & global_context_,
+            const Poco::Timespan & http_timeout_,
+            const std::string & connection_string_)
+    : log(&Poco::Logger::get(BridgeHelperMixin::getName() + "BridgeHelper"))
+    , connection_string(connection_string_)
+    , http_timeout(http_timeout_)
+    , context(global_context_)
+    , config(context.getConfigRef())
+{
+    bridge_host = config.getString(BridgeHelperMixin::configPrefix() + ".host", DEFAULT_HOST);
+    bridge_port = config.getUInt(BridgeHelperMixin::configPrefix() + ".port", DEFAULT_PORT);
+}
+
+
+protected:
+    auto getConnectionString() const { return connection_string; }
+
+    const String getName() const override { return BridgeHelperMixin::getName(); }
+
+    size_t getDefaultPort() const override { return DEFAULT_PORT; }
+
+    const String serviceAlias() const override { return BridgeHelperMixin::serviceAlias(); }
+
+    /// Same for odbc and jdbc
+    const String serviceFileName() const override { return "clickhouse-odbc-bridge"; }
+
+    const String configPrefix() const override { return BridgeHelperMixin::configPrefix(); }
+
+    const Context & getContext() const override { return context; }
+
+    const Poco::Timespan & getHTTPTimeout() const override { return http_timeout; }
+
+    const Poco::Util::AbstractConfiguration & getConfig() const override { return config; }
+
+    Poco::Logger * getLog() const override { return log; }
+
+    bool startBridgeManually() const override { return BridgeHelperMixin::startBridgeManually(); }
+
+    Poco::URI createBaseURI() const override
+    {
+        Poco::URI uri;
+        uri.setHost(bridge_host);
+        uri.setPort(bridge_port);
+        uri.setScheme("http");
+        return uri;
+    }
+
+    void startBridge(std::unique_ptr<ShellCommand> cmd) const override
+    {
+        context.addBridgeCommand(std::move(cmd));
+    }
+
+
+private:
+    using Configuration = Poco::Util::AbstractConfiguration;
+
+    Poco::Logger * log;
+    std::string connection_string;
+    const Poco::Timespan & http_timeout;
+    std::string bridge_host;
+    size_t bridge_port;
+
+    const Context & context;
+    const Configuration & config;
+
+    std::optional<IdentifierQuotingStyle> quote_style;
+    std::optional<bool> is_schema_allowed;
+
+
+protected:
+    Poco::URI getColumnsInfoURI() const override
+    {
+        auto uri = createBaseURI();
+        uri.setPath(COL_INFO_HANDLER);
+        return uri;
+    }
+
+    std::vector<std::pair<std::string, std::string>> getURLParams(
```

Comment:
Probably it's not your code, but this type is ugly. Need `using` with a clear name.